### ISSUE TITLE
bug: Wrong message body format on SQS message received through ReceiveMessage action.

### ISSUE THREAD

**Issue by SmashingQuasar** [localstack/localstack#8451]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When sending a message with a JSON body, SQS does not respond with the correct body when called through raw HTTP.

For example:

```bash
awslocal sqs send-message --queue-url http://localhost:4566/000000000000/my-queue --message-body '{"foo": "bar"}'
```

Will create a message with the following JSON body:

```json
{"foo": "bar"}
```

When calling:

```bash
awslocal sqs receive-message --queue-url http://localhost:4566/000000000000/my-queue
```

You will receive the corresponding message as:

```json
{
    "Messages": [
        {
            "MessageId": "edee64b2-0ee0-44ac-803e-11c902e7d759",
            "ReceiptHandle": "MDY1ZmI3ZDUtNTU1OS00MDY2LTg4YzgtOGE5YTY2YTRmNDMxIGFybjphd3M6c3FzOnVzLWVhc3QtMTowMDAwMDAwMDAwMDA6am9iX3NvdXJjZV9hdWRpb19zdGF0dXMgZWRlZTY0YjItMGVlMC00NGFjLTgwM2UtMTFjOTAyZTdkNzU5IDE2ODYxNDEyMjQuNzAyMzUxNg==",
            "MD5OfBody": "9bb58f26192e4ba00f01e2e7b136bbd8",
            "Body": "{\"foo\":\"bar\"}"
        }
    ]
}
```

This only works when using the CLI.

If you send a message through SQS directly through HTTP using Postman (or cURL, whatever you are more comfortable with), you will end up with a malformed answer such as:

```json
{
    "ReceiveMessageResponse": {
        "ReceiveMessageResult": {
            "Message": {
                "MessageId": "8c20c53e-1877-4da1-9fec-f64ac8a2b494",
                "ReceiptHandle": "ZDk2OWZjNDMtZTQ0Ny00MjQyLWJkZjYtZmNmMzEyNjAzNzBiIGFybjphd3M6c3FzOnVzLWVhc3QtMTowMDAwMDAwMDAwMDA6am9iX3NvdXJjZV9hdWRpb19zdGF0dXMgOGMyMGM1M2UtMTg3Ny00ZGExLTlmZWMtZjY0YWM4YTJiNDk0IDE2ODYxNDEyNTIuODE4NjY5Ng==",
                "MD5OfBody": "94232c5b8fc9272f6f73a1e36eb68fcf",
                "Body": "{__marker__\"__marker__foo__marker__\"__marker__: __marker__\"__marker__bar__marker__\"__marker__}"
            }
        },
        "ResponseMetadata": {
            "RequestId": "9633f6f0-100f-4ad2-8292-fe63e87b17f4"
        }
    }
}
```

### Expected Behavior

SQS should return the body as it was sent, not altered with `__marker__` tags.

This issue seemingly comes from the following commit: https://github.com/localstack/localstack/commit/eff7f2c7c5bffcde65c1c91529749a22e80e48b3

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

To reproduce this issue, I recommend using the official Localstack image on it's latest tag.

Start LocalStack with SQS enabled and create any queue by running within your LocalStack container.

```bash
awslocal sqs create-queue --queue-name sample-queue
```

Then, send a message to the created queue:

```bash
awslocal sqs send-message --queue-url http://localhost:4566/000000000000/sample-queue --message-body '{"foo": "bar"}'

# or

curl -k 'http://localhost:4566/000000000000/sample-queue?Action=SendMessage&MessageBody={%22foo%22%3A%22bar%22}' \
--header 'Accept: application/json' \
--header 'Version: 2012-11-05'
```

Now get the message with the CLI:

```bash
awslocal sqs receive-message --queue-url http://localhost:4566/000000000000/sample-queue
```

This will return the correct message structure.

Now with cURL:

```bash
curl 'https://localhost:4566/000000000000/sample-queue?Action=ReceiveMessage&MaxNumberOfMessage=5&VisibilityTimeout=15&AttributeName=All' \
--header 'Accept: application/json'
```

This will return the message with the `__marker__` delimiters still present.

### Environment

```markdown
- OS: Debian 11 Bullseye
- LocalStack: latest
```

**Comment by SmashingQuasar** [comment]
After some further testing with previous LocalStack versions, it appears LocalStack 1.1 was already problematic.

Where live AWS answers with `"Body":"{\"foo\":\"bar\"}"`, LocalStack answers with `"Body": "{&quot;foo&quot;:&quot;bar&quot;}"` with the exact same entry request sent through cURL.

Maybe it would be wiser to not alter the message body whatsoever within LocalStack? It looks like more trouble than it's worth.

**Comment by SmashingQuasar** [comment]
I tried using `POST` as a workaround but the same problem exists with `POST`.

**Comment by SmashingQuasar** [comment]
This appears to affect other endpoints as well.
For example the `DeleteMessage` action will return a string with `__marker__` present when an invalid ReceiptHandle parameter is given.
![image](https://github.com/localstack/localstack/assets/11139591/2f8748e6-117d-4cba-9289-fbfef3650606)

**Comment by SmashingQuasar** [comment]
> thanks for reporting Nicolas @SmashingQuasar, glad you found a workaround for now. We do know that this is a limitation right now, unfortunately there's not much we can do. see related discussions:
> 
>     * [bug: &quot; when sending replaced by quotes when receiving #8002](https://github.com/localstack/localstack/issues/8002)
> 
>     * [fix invalid SQS message decoding #8180](https://github.com/localstack/localstack/pull/8180)

Sorry for the late response, I was on vacations.

I understand the issue you were trying to address, however I fail to understand what prevents you from removing the `__marker__` that you added within the message when sending it back. If you can add them, then surely you can remove them, this all sounds like an internal parsing issue that does not seem related to any of the input given by the consumers.

**Comment by thrau** [comment]
i did a bit of digging and also confirmed the issue. i think we should probably tackle this once #9710 is merged, since it also affects the serialization of json vs xml responses.

there's also the aspect of the query API which should also support json

**Comment by ekivolowitz** [comment]
I'm also running into this issue as well. I fully agree with @SmashingQuasar. I'm using localstack for automated testing and I don't want introduce code in the hot path of production that will handle removing `__marker__`.

I see that #9710  is merged in. @thrau is it possible to raise the priority of this? I expect that it's a somewhat common usecase.

**Comment by SmashingQuasar** [comment]
@thrau What's the current status of this issue?
It seems like a fairly trivial problem to fix. If you need someone to handle it I can have a look at it.

**Comment by baermat** [comment]
Hello @SmashingQuasar 
I will take a look at it this week actually, and hopefully it will be resolved. I will post further updates once I have more information.

**Comment by thrau** [comment]
Should now be fixed with #10732

**Comment by danilobassi8** [comment]
Hi, What's the current status of this?
I'm using the latest localstack docker image and i'm having the same issue.
